.evergreen/run-fuzz.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+set -o errexit  # Exit the script with error if any of the commands fail
+
+FUZZTIME=10m
+
+# Change the working directory to the root of the mongo repository directory
+cd $PROJECT_DIRECTORY
+
+# Get all go test files that contain a fuzz test.
+FILES=$(grep -r --include='**_test.go' --files-with-matches 'func Fuzz' .)
+
+# For each file, run all of the fuzz tests in sequence, each for -fuzztime=FUZZTIME.
+for FILE in ${FILES}
+do
+	PARENTDIR="$(dirname -- "$FILE")"
+
+	# Clean testing cache.
+	go clean -testcache
+	go clean -fuzzcache
+
+	# Get a list of all fuzz tests in the file.
+	FUNCS=$(grep -o 'func Fuzz[A-Za-z0-9]*' $FILE | cut -d' ' -f2)
+
+	# For each fuzz test in the file, run it for FUZZTIME.
+	for FUNC in ${FUNCS}
+	do
+		echo "Fuzzing \"${FUNC}\" in \"${FILE}\""
+
+		# Create a set of directories that are already in the subdirectories testdata/fuzz/$fuzzer corpus. This
+		# set will be used to differentiate between new and old corpus files.
+		declare -a cset
+
+		if [ -d $PARENTDIR/testdata/fuzz/$FUNC ]; then
+			# Iterate over the files in the corpus directory and add them to the set.
+			for SEED in $PARENTDIR/testdata/fuzz/$FUNC/*
+			do
+				cset+=("$SEED")
+			done
+		fi
+
+		go test ${PARENTDIR} -run=${FUNC} -fuzz=${FUNC} -fuzztime=${FUZZTIME} || true
```

Comment:
Do we not care about the output of `go test` (see the `|| true`)?

Reply:
[This will still pipe the output](https://stackoverflow.com/a/11231970), but if the `go test` panics/errors due to a crash (which is good in the case) we don't want the bash script to terminate, rather continue the generated corpus processing.

Reply:
Sounds good, thanks

Reply:
@matthewdale Yes, this should still output the results of the test. If I added a `panic` to the fuzz test and then added checkpoints to the script like this:

```sh
echo "before"
go test ${PARENTDIR} -run=${FUNC} -fuzz=${FUNC} -fuzztime=${FUZZTIME} || true
echo "after"
```

The output would be 

```
Fuzzing "FuzzDecode" in "./bson/fuzz_test.go"
before
fuzz: elapsed: 0s, gathering baseline coverage: 0/460 completed
failure while testing seed corpus entry: FuzzDecode/seed#0
fuzz: elapsed: 0s, gathering baseline coverage: 0/460 completed
--- FAIL: FuzzDecode (0.08s)
    --- FAIL: FuzzDecode (0.00s)
        testing.go:1356: panic: test
            goroutine 60 [running]:
            runtime/debug.Stack()
                /opt/homebrew/Cellar/go/1.19.1/libexec/src/runtime/debug/stack.go:24 +0x104
            testing.tRunner.func1()
                /opt/homebrew/Cellar/go/1.19.1/libexec/src/testing/testing.go:1356 +0x258
            panic({0x104d9f300, 0x104e0eac0})
                /opt/homebrew/Cellar/go/1.19.1/libexec/src/runtime/panic.go:884 +0x204
            go.mongodb.org/mongo-driver/bson.FuzzDecode.func1(0x1400013d718?, {0x10476bb28?, 0x0?, 0x0?})
                /Users/preston.vasquez/Developer/mongo-go-driver/bson/fuzz_test.go:19 +0xb8
            reflect.Value.call({0x104da5400?, 0x104e0d9c0?, 0x13?}, {0x104cc889d, 0x4}, {0x140002d46c0, 0x2, 0x2?})
                /opt/homebrew/Cellar/go/1.19.1/libexec/src/reflect/value.go:584 +0x688
            reflect.Value.Call({0x104da5400?, 0x104e0d9c0?, 0x1400008c820?}, {0x140002d46c0?, 0x0?, 0x1400032e978?})
                /opt/homebrew/Cellar/go/1.19.1/libexec/src/reflect/value.go:368 +0x90
            testing.(*F).Fuzz.func1.1(0x1400007eba0?)
                /opt/homebrew/Cellar/go/1.19.1/libexec/src/testing/fuzz.go:337 +0x1dc
            testing.tRunner(0x14000221520, 0x140002d8480)
                /opt/homebrew/Cellar/go/1.19.1/libexec/src/testing/testing.go:1446 +0x10c
            created by testing.(*F).Fuzz.func1
                /opt/homebrew/Cellar/go/1.19.1/libexec/src/testing/fuzz.go:324 +0x4c4


FAIL
exit status 1
FAIL    go.mongodb.org/mongo-driver/bson        0.298s
after
```